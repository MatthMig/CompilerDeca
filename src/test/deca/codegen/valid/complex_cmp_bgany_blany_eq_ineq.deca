// Description :
// 
//      Test complexe de différents opérateurs arithmetiques et de comparaisons
// 
// Résultats : 
// 
//      N/A
// 
// Historique : 
// 
//      Créé le 09/06/2023 par gl03

{
    float x=0;
    float y=10;
    while (((x<=1)||(x<=2*2))||((x<10/1)&&(x<11%12)))
    {
        y=10;
        while (((y>=10)||(y>9.0))||(y>=0)&&(y>0))
        {
            if(x==y)
            {
                println("hello");
                println(x);
                println(y);
            }
            y=y-1;
        }
        x=x+1;
    }
    x=x+1.8;
    y=y+1.8;
    if(x!=y)
    {
        println("bye");
    }
}