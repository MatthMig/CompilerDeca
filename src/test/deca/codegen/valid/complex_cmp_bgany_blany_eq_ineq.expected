hello
1.00000e+00
1.00000e+00
hello
2.00000e+00
2.00000e+00
hello
3.00000e+00
3.00000e+00
hello
4.00000e+00
4.00000e+00
hello
5.00000e+00
5.00000e+00
hello
6.00000e+00
6.00000e+00
hello
7.00000e+00
7.00000e+00
hello
8.00000e+00
8.00000e+00
hello
9.00000e+00
9.00000e+00
bye
